ConfigurationSystem/private/ConfigurationClient.py

```diff
@@ -146,20 +233,44 @@ def getOptionsDict(self, sectionPath):
     else:
       return S_ERROR("Path %s does not exist or it's not a section" % sectionPath)
 
-  def getConfigurationTree(self, root='', *filters):
-    """
-    Create a dictionary with all sections, subsections and options
-    starting from given root. Result can be filtered.
+  def getOptionsDictRecursively(self, sectionPath):
+    """ Get configuration options in dictionary recursively
 
-    :param str root: Starting point in the configuration tree.
-    :param filters: Select results that contain given substrings (check full path, i.e. with option name)
-    :type filters: str or python:list[str]
-    :return: Return a dictionary where keys are paths taken from
-             the configuration (e.g. /Systems/Configuration/...).
-             Value is "None" when path points to a section
-             or not "None" if path points to an option.
+        :param basestring sectionPath: section path
+
+        :return: S_OK(dict)/S_ERROR()
     """
+    optionsDict = {}
```

Comment:
Can't you use this function?
https://github.com/DIRACGrid/DIRAC/blob/2a02b781e40c4c11c027e1d3bf92b398ad908d75/Core/Utilities/CFG.py#L460

I think this should be equivalent to your implementation
```python
def getOptionsDictRecursively(self, sectionPath):
  if not gConfigurationData.mergedCFG.isSection(sectionPath):
    return S_ERROR("Path %s does not exist or it's not a section" % sectionPath)
  return S_OK(gConfigurationData.mergedCFG.getAsDict(sectionPath))
```

Reply:
Yes, thanks. Looks much better.